nullaway/src/test/java/com/uber/nullaway/jspecify/GenericsTests.java

```diff
@@ -1856,6 +1856,52 @@ public void issue1008() {
         .doTest();
   }
 
+  @Test
+  public void intersectionTypeFromConditionalExprInStringConcat() {
+    makeHelper()
+        .addSourceLines(
+            "ServiceExtraInfo.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "public class ServiceExtraInfo {",
+            "    private java.util.@Nullable List<Object> relativeServices;",
+            "    private String getStr() {",
+            "        return (relativeServices == null ? \"relativeServices == null\" : relativeServices.size()) + \"\";",
+            "    }",
+            "    private String getStr2(boolean b, java.util.List<Object> l) {",
+            "        return (b ? (b ? l.size() : \"hello\") : 3) + \"\";",
+            "    }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void intersectionTypeInvalidAssign() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "import java.io.Serializable;",
+            "public class Test {",
+            "  interface A<T extends @Nullable Object> {}",
+            "  static class B implements A<@Nullable String>, Serializable {}",
+            "  static void test1(Object o) {",
+            "    var x = (A<String> & Serializable) o;",
+            "    // BUG: Diagnostic contains: Cannot assign from type B to type A<String> & Serializable",
+            "    x = new B();",
+            "  }",
```

Comment:
To be clear, the case where we have:

```
            "    var x = (A<@Nullable String> & Serializable) o;",
            "    x = new B();",
```

Should have no error, right? Should we encode that test case just to make sure we never regress that?

Reply:
you're right, and i think we will get a false positive on that test for now (since for some reason `var`-declared variables don't get types with all the right annotations).  i'll add tests and open an issue for outstanding tasks, but will fix in a subsequent PR

Reply:
Addressed in 5a8bdd6393f0d6b2c54801b420dccdb46bd72b17 along with link to issue https://github.com/uber/NullAway/issues/1022